### PR TITLE
add more exiftool examples

### PR DESCRIPTION
**Pull request by therealmarv**

### CODE REVIEW

**Comment by rubenvereecken** [comment]
Hi, thanks for taking the time to come up with some more use cases!

I've got some questions though, as I've only used the tool once myself
- Is manipulating timestamps in the way you propose really one of the most common use cases of the tool? It sounds like a strong metadata tool, and in that case changing time is only a small feature and not very representative. If it is though, do tell.
- Is the verbose option necessary to make the wildcard work in your second example? Or does it just take care of verbose output - in that case it shouldn't be added without describing it. Sorry, I'm on a Chromebook for the moment.

If you can get back to me on those things we can make this work :).

**Comment by therealmarv** [comment]
1. practically speaking, yes. Changing timestamps is one of the most common use cases. Imagine that you've shot a bunch of pictures but  e.g. did not adjusted summertime or you have taken pictures in a foreign country. Especially when more then one camera is involved (vacation with friends etc.).
2. No the verbose is not necessary. I will remove it.

About what you can do with exiftool: It is of course more powerful and can do a lot of more things. This two examples are just two which are used very often with exiftool.

**Comment by waldyrious** [comment]
FWIW, that's what I have used exiftool for, too. I had pictures from the same event but from different cameras which were either wildly or slightly unsynchronized, and exiftool allowed me to make the timestamps compatible so the photos could be navigated in chronological order, and rename the various filename standards to a common timestamp-based format.

Here are the personal notes I kept after I learned how to use the tool, feel free to adapt any of this into your PR if you think it's worth it:

> How to: Adjust “date photo taken” timestamp for all images in a directory:

> `sudo apt-get install libimage-exiftool-perl`

> `exiftool -DateTimeOriginal-='0:0:0 1:30:0' <photos>`
> (or alternatively, `-DateTimeOriginal-=1.5` -- default unit is hours)

> The “photos” parameter can be a directory, a file, a list of files, or a wildcard expression. You can recursively parse the contents of a directory with the `-r` option.

> The option `-overwrite_original` (or `-overwrite_original_in_place` to preserve filesystem attributes such as creation/modification date) can be used to prevent the creation of backups (same filenames, with “_original” appended). Additionally, the `-o` (output) option can be used to specify a different filename for the output (it can be combined with the overwrite option to essentially rename the file). The output filename can use the placeholders `%d` (directory),` %f` (filename), `%e` (extension), and more.

> Here's an example of renaming the files:

> `exiftool '-filename<DateTimeOriginal' -d %Y-%m-%d_%H-%M-%S%%lc.%%e </path/to/directory/> -r -ext jpg`

> Note: I got several “Error: [minor] MakerNotes offsets may be incorrect (fix or ignore?)” when attempting this. To choose "fix", use the `-F` option; to choose "ignore", use `-m`.

**Comment by rubenvereecken** [comment]
Thanks for leaving out `verbose`. And thanks @waldyrious to add your own notes here. I'll leave it up to @therealmarv to see if he wants to incorporate any of it. I do see some rather interesting options in there, like `-r` and `-ext` (that are apparently even preferred over filename wildcards).

**Comment by therealmarv** [comment]
I've added some more options thanks to the helpful comments here. Exiftool is very powerful. Only thing I used it so far is removing exif information or shifting times of exif.

**Comment by waldyrious** [comment]
IIRC `AllDates` changes also file creation and modification dates, which I'm not sure will be intended for most users manipulating EXIF data. `DateTimeOriginal` sounds like the right field to edit, as it represents "the date and time the picture was taken or recorded" (quoting [this page](http://www.awaresystems.be/imaging/tiff/tifftags/privateifd/exif/datetimeoriginal.html)).

(By the way, [this other page](http://blog.chillantro.com/appsperience/2012/08/22/what-exif-date-and-time-tag-in-images-is-date-time-vs-date-time-original-vs-date-time-digitized/) goes into detail about the difference between `DateTimeOriginal` and `DateTimeDigitized` (e.g. for scanned photos). IMO that distinction is too obscure for the most basic/common usage, where digital pictures can be assumed.)

That said, I do read in the [documentation](http://www.sno.phy.queensu.ca/~phil/exiftool/#shift) that:

> Commonly, in JPEG images, the DateTimeOriginal, CreateDate and ModifyDate values must all be changed.

Do you know why that is so? Is that due to limitations in image managers?

If possible, I'd prefer if we used only one tag for date shift, to keep things simple.

**Comment by therealmarv** [comment]
Yes, you are not the only one who is confused. You're right that the image manager programmers are also not conform to one EXIF field (creation, modification etc). Also keep in mind when the camera timer is wrong all date fields are wrong including file creation and modification date and DateTimeOriginal.

So it is common practice to change every field to match eachother. With this approach you are on the safe side.

I also do not like the spec... it even does not cover time zones really good. Common practice is to set it to the local time where the picture is taken but there are also timezone fields which nobody is using.

So tl;dr
Change every date and let them match to have no problems.

**Comment by waldyrious** [comment]
> Also keep in mind when the camera timer is wrong all date fields are wrong including file creation and modification date and DateTimeOriginal.

I'm not sure this is actually the case. I'd assume these are modified e.g. when photos are edited (and these days with free tools available like Picasa, Windows Photo Editor, Photoshop Express, etc., it's very common even for non-tech-savvy users), while DateTimeOriginal likely would remain as it was.

I would therefore prefer not to induce people to change these dates unwittingly and potentially lose metadata about their files, for the sake of working-around bugs in image viewers. In fact, come to think about it, I am not even aware of any such viewer that shows this bug -- I certainly didn't have a problem with either the default viewers in Windows and Ubuntu when I last tried this -- editing only DateTimeOriginal was sufficient.

**Comment by therealmarv** [comment]
As I said it is complicated. You may only change DateTimeOriginal.

So the problem is... we have too many fields. In EXIF [(and in worst case maybe even XMP, IPTC & Composite)](https://forums.adobe.com/message/6689065#6689065) we have:

* Date Created (don't now the exact tag name)
* Date Modified (also don't now the exact tag name)
* DateTimeOriginal
* DateTimeDigitized?!
* maybe even more?!

and then in the file object itself we have:

* Date created
* Date modified
* Last view (Windows?)

So if you want to do it 100% correctly you have to look at at least 6 different date objects.

So when you have a wrong camera timer and you only change DateTimeOriginal and at least Date Created and Date Modified will stay wrong. I cannot speak for all but when my images are wrong sorted I do not want to keep care of 6 or 8 fields of metadata and 6 or 8 parameters. This is why the parameter `-AllDates` exists and I'm happy it is easy this way.

See also the answers here for further discussion:
http://photo.stackexchange.com/questions/7919/how-to-shift-exif-date-time-created-by-time-in-days-hours-minutes

As I said I also do not like the spec but AllDates will keep sure you modified most of it right for right order of photos. If you want to suggest more complicated exiftool commands with changing 3-4 fields without changing the other fields you are more than welcome.

I will think of changing the tldr text describing alldates command...

**Comment by waldyrious** [comment]
I would only agree in principle with AllDates if the tool somehow could detect whether they are all the same and only change them all if so. If any modifications (save with new name, add filters, etc.) have occurred, thus changing the DateCreated and DateModified, we should encourage preserving this metadata.

That said, if we can explain the above very concisely in the descriptions of the examples including AllDates, then I'd support retaining it. Otherwise, users would be left guessing what AllDates includes (and why they would want to use it) as opposed to DateTimeOriginal. As you can see, conciseness is not my strength :P looking forward to see what you can come up with!

**Comment by denis-sokolov** [comment]
Is it not the case that we do not need to worry about backwards compatibility in TL;DR and thus can, instead, accept examples as long as there are up to 5 for a page? Then, if we want more examples, it will be easier to choose which examples to leave and which to remove.

**Comment by rubenvereecken** [comment]
On a cleaning spree here.  @therealmarv, any chance you're still around? @waldyrious, feel like you still have anything to add? 

We could hijack this PR if marv can't be tracked and it adds value.

**Comment by ostera** [comment]
@therealmarv  @rubenvereecken, I hijacked this D:

Format fixes here: https://github.com/tldr-pages/tldr/commit/c701cca93054295a84d6735215ad1e9148bb2310
